ts/Series/Pictorial/PictorialSeries.ts

```diff
@@ -0,0 +1,716 @@
+                    }
+                })
+                .add(stack.shadowGroup);
+
+            invertShadowGroup(
+                stack.shadowGroup,
+                xAxis,
+                stack.axis
+            );
+
+            rescalePatternFill(
+                stack.shadow,
+                height,
+                width,
+                height,
+                strokeWidth
+            );
+
+            stack.setOffset(
+                series.pointXOffset || 0,
+                series.barW || 0
+            );
+
+        } else if (stackShadow && stack.shadowGroup) {
+            stackShadow.animate({
+                x,
+                y,
+                width,
+                height
+            });
+            const fillUrlMatcher = /url\(([^)]+)\)/,
+                fill = stackShadow.attr('fill') as string,
+                match = fill && fill.match(fillUrlMatcher);
+
+            if (match && chart.renderer.patternElements) {
+                chart.renderer.patternElements[match[1].slice(1)]
+                    .animate({
+                        x,
+                        y,
+                        width,
+                        height
+                    });
+            }
+            stack.shadowGroup.animate({
+                translateX: chart.inverted ?
+                    stack.axis.pos : xAxis.pos,
+                translateY: chart.inverted ?
+                    xAxis.pos : stack.axis.pos
+            });
+
+            invertShadowGroup(
+                stack.shadowGroup,
+                xAxis,
+                stack.axis
+            );
+
+            rescalePatternFill(
+                stackShadow,
+                height,
+                width,
+                height,
+                strokeWidth
+            );
+
+            stack.setOffset(
+                series.pointXOffset || 0,
+                series.barW || 0
+            );
+
+        }
+    } else if (stack.shadow && stack.shadowGroup) {
+        stack.shadow.destroy();
+        stack.shadow = void 0;
+
+        stack.shadowGroup.destroy();
+        stack.shadowGroup = void 0;
+    }
+}
+
+function forEachStack(chart: Chart, callback: Function): void {
+    if (chart.axes) {
+        chart.axes.forEach(function (axis): void {
+            if (!axis.stacking) {
+                return;
+            }
+
+            const stacks = axis.stacking.stacks;
+            // Render each stack total
+            objectEach(stacks, function (
+                type: Record<string, StackItem>
+            ): void {
+                objectEach(type, function (stack: StackItem): void {
+                    callback(stack);
+                });
+            });
+        });
+    }
+}
+
+addEvent(Chart, 'render', function (): void {
+    forEachStack(this, renderStackShadow);
+});
+
+addEvent(StackItem, 'afterSetOffset', function (e): void {
+    if (this.shadow) {
+        this.shadow.attr({
+            translateX: (e as any).xOffset
```

Comment:
We can create an interface for the event arguments to avoid any casting. See for example [GetSelectionMarkerAttrsEvent](https://github.com/highcharts/highcharts/blob/5c7bcfd144199b4c798ebca6080dc044126aee1c/ts/Core/Pointer.ts#L288).